packages/cli/sw/sw.js

```diff
@@ -0,0 +1,4 @@
+import { getFiles, setupPrecaching, setupRouting } from './index';
+
+setupRouting();
```

Comment:
I'm wondering if we're locking ourselves into too strict an API here with the two functions. Thoughts on this?

```
import { getFiles, init } from './index';

setup({
  files: getFiles(),
  // other options here
});
```

Reply:
Coincidently that was literally my initial design.
The problem i found with this design was that `precaching` and `routing` are 2 very independent pieces.

The problem that we're trying to solve(have a large base of basic users which just use our API and we can change under lying implementation for them without a breaking change) fades away here because
- Pretty much everyone uses precaching
- We introduce a fallback page for error/offline cases
- We have a default strategy for navigation request(Network first currently)

If my app changes any of the above(e.g. a game or a music app can be/should be cache first or i might want to change my catch behavior to a "you are offline" page) then i'll need to do everything inside `init` again.

vs

In the currently implemented design, I can skip routing and just `precaching`

Reply:
TBH based on that I'm fine with leaving them separate.